satpy/readers/ami_l1b.py

```diff
@@ -212,3 +262,32 @@ def _calibrate_ir(self, dataset_id, data):
             bt_data = c0 + c1 * t_eff + c2 * t_eff * t_eff
             data.data = bt_data
         return data
+
+    def _get_rad_corr_factors(self):
+        """Retrieve radiance correction factors from user-supplied dict."""
+        try:
+            slope = self.radiance_correction[self.band_name]['slo']
+            offset = self.radiance_correction[self.band_name]['off']
+        except KeyError:
+            raise KeyError("You have selected to supply radiance correction "
+                           "factors but have not included any for channel " +
+                           self.band_name)
+        return slope, offset
+
+    def _apply_gsics_rad_correction(self, data):
+        """Retrieve GSICS factors from L1 file and apply to radiance."""
+        rad_slope = self.nc['gsics_coeff_slope'][0]
+        rad_offset = self.nc['gsics_coeff_intercept'][0]
+        data = self._apply_rad_correction(data, rad_slope, rad_offset)
+        return data
+
+    def _apply_user_rad_correction(self, data):
+        """Retrieve user-supplied radiance correction and apply."""
+        rad_slope, rad_offset = self._get_rad_corr_factors()
+        data = self._apply_rad_correction(data, rad_slope, rad_offset)
+        return data
+
+    def _apply_rad_correction(self, data, slo, off):
```

Comment:
This method could be shared by both readers.

Reply:
Done.